File: java8-scala/src/java8/Main03.scala
package java8

/** An example of lambdas. */
object Main03 {
  def main(args: Array[String]) {
    val list = Seq("1", "2", "3", "44")
    
    val lengths2 = list.map { _.length }
    println(lengths2)
    
    val bigLengths2 = list.filter { _.size > 1 }.map { _.length }
    println(bigLengths2)
  }
}

File: java8/src/java8/Main03.java
package java8;

import static java.util.stream.Collectors.toList;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/** Example of lambdas. */
public class Main03 {

  public static void main(String[] args) {
    List<String> list = Arrays.asList("1", "2", "3", "44");

    List<Integer> lengths = new ArrayList<>();
    for (String e : list) {
      lengths.add(e.length());
    }
    System.out.println(lengths);

    List<Integer> lengths2 = list.stream().map(e -> e.length()).collect(toList());
    System.out.println(lengths2);

    List<Integer> bigLengths = new ArrayList<>();
    for (String e : list) {
      if (e.length() > 1) {
        bigLengths.add(e.length());
      }
    }
    System.out.println(bigLengths);

    List<Integer> bigLengths2 = list.stream()//
      .filter(e -> e.length() > 1)
      .map(e -> e.length())
      .collect(toList());
    System.out.println(bigLengths2);
  }

}
